sycl/include/CL/sycl/kernel.hpp

```diff
@@ -8,16 +8,18 @@
 
 #pragma once
 
-#include <CL/sycl/detail/kernel_impl.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/info/info_desc.hpp>
 #include <CL/sycl/stl.hpp>
 
-#include <memory>
```

Comment:
I believe that memory header is needed as shared_ptr is used in the file.

Reply:
I put it back